carla-simulator/carla
# Carla spectator

Carla Spectator added ,changes related to game mode, controllers, vehicles, hud, etc.

Checklist:

  - [x] Add a class to be used as default spectator pawn of the game mode for CARLA
  - [x] Add controls in the settings and code to control the the spectator pawn (key E)
  - To test
    - Open a map, create a blueprint that inherits from CarlaSpectator, add a mesh to it to see the spectator
    - Set this new spectator blueprint as default spectator class in the game mode
    - Run the game ,use the E key to exit and enter cars

#### Description

This is the branch to allow another pawn different from a vehicle to control the camera

child of #438 

#### Where has this been tested?

  * **Platform(s):** Windows and Linux
  * **Unreal Engine version(s):** 4.18.3

#### Possible Drawbacks
Problems related to the assumption that the controller has to control a vehicle pawn


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/carla-simulator/carla/445)
